pkg/analysis/analyzer.go

```diff
@@ -6,16 +6,18 @@ import (
 	"time"
 
 	"github.com/pkg/errors"
+	"go.nanomsg.org/mangos/v3/protocol"
 	"nodemon/pkg/entities"
 	"nodemon/pkg/storing/events"
 )
 
 type Analyzer struct {
-	es *events.Storage
+	es     *events.Storage
+	socket protocol.Socket
 }
 
-func NewAnalyzer(es *events.Storage) *Analyzer {
-	return &Analyzer{es: es}
+func NewAnalyzer(es *events.Storage, socket protocol.Socket) *Analyzer {
```

Comment:
No need to pass `socket` to `Analyzer`, it's already has channel that produces Alerts.

Reply:
I still need a place to send messages to socket

Reply:
Do it in another goroutine. Use channels for that.